### PR TITLE
cockpit: 284 -> 287 + update script

### PR DESCRIPTION
**Pull request by lucasew**
###### Description of changes
- Bump cockpit to 287
- Changes to the patch logic, so it will work in a more generic way, hopefully avoiding that line shifts break the build
- Update script so r-ryantm can submit the next update PR

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lucasew** [comment]
Cockpit itself is building successfully, but Firefox that is a dependency for the integration test cache missed.

**Comment by lucasew** [comment]
@ofborg build cockpit.tests.cockpit

File: pkgs/servers/monitoring/cockpit/default.nix [hunk 1]
```diff
@@ -88,20 +88,26 @@ stdenv.mkDerivation rec {
...
-  ];
+    substituteInPlace src/systemd/Makefile.am \
+      --replace 'mkdir -p $(DESTDIR)$(sysconfdir)/motd.d' 'mkdir -p $(DESTDIR)$(prefix)$(sysconfdir)/motd.d' \
+      --replace 'ln -sTfr $(DESTDIR)/run/cockpit/motd $(DESTDIR)$(sysconfdir)/motd.d/cockpit' 'ln -sTfr $(DESTDIR)$(prefix)/run/cockpit/motd $(DESTDIR)$(prefix)$(sysconfdir)/motd.d/cockpit'  \
+      --replace 'mkdir -p $(DESTDIR)$(sysconfdir)/issue.d' 'mkdir -p $(DESTDIR)$(prefix)$(sysconfdir)/issue.d' \
+      --replace 'ln -sTfr $(DESTDIR)/run/cockpit/motd $(DESTDIR)$(sysconfdir)/issue.d/cockpit.issue' 'ln -sTfr $(DESTDIR)$(prefix)/run/cockpit/motd $(DESTDIR)$(prefix)$(sysconfdir)/issue.d/cockpit.issue'
+
+    substituteInPlace src/ws/Makefile-ws.am \
+      --replace 'mkdir -p $(DESTDIR)$(sysconfdir)/cockpit/ws-certs.d $(DESTDIR)$(sysconfdir)/cockpit/machines.d' 'mkdir -p $(DESTDIR)$(prefix)$(sysconfdir)/cockpit/ws-certs.d $(DESTDIR)$(prefix)$(sysconfdir)/cockpit/machines.d' \
+      --replace 'chmod 755 $(DESTDIR)$(sysconfdir)/cockpit/ws-certs.d $(DESTDIR)$(sysconfdir)/cockpit/machines.d' 'chmod 755 $(DESTDIR)$(prefix)$(sysconfdir)/cockpit/ws-certs.d $(DESTDIR)$(prefix)$(sysconfdir)/cockpit/machines.d'
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/cockpit/default.nix:104]: Can you try to upstream this?

File: pkgs/servers/monitoring/cockpit/default.nix [hunk 2]
```diff
@@ -215,7 +221,10 @@ stdenv.mkDerivation rec {
     npm run stylelint
   '';
 
-  passthru.tests = { inherit (nixosTests) cockpit; };
+  passthru = {
+    tests = { inherit (nixosTests) cockpit; };
+    updateScript = ./update.sh;
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/cockpit/default.nix:226]:
```suggestion
```
the bot will use nix-update which does the same thing

Comment (lucasew) [reply pkgs/servers/monitoring/cockpit/default.nix:226]: I didn't know about that

BTW, this bot is a bit obscure IMO, so I wanted to be a bit more explicit about the update behavior.

File: pkgs/servers/monitoring/cockpit/default.nix (hunk 1 above)
Comment (lucasew) [reply pkgs/servers/monitoring/cockpit/default.nix:104]: yes https://github.com/cockpit-project/cockpit/pull/18513

Comment (AndersonTorres) [reply pkgs/servers/monitoring/cockpit/default.nix:104]: Use this patch file you upstreamed instead of the script-patch above.

Comment (lucasew) [reply pkgs/servers/monitoring/cockpit/default.nix]: This broke their build

I tried another approach, just editing the relevant references. Tested in my daily driver and it's working.

**Pull request merged**